### DISCUSSION TITLE
Parameters from Delta 2 Max

### DISCUSSION THREAD

**Discussion by Neewieedo** [tolwi/hassio-ecoflow-cloud#277] [Ideas]
Hi, 
I have two questions/ideas about the parameters of the Delta 2 Max.

1. Do I understand correctly that you can set the AC charging speed using the entity _number.delta_2_max_ac_charging_power_ and does that work for you? I always get 2400W displayed in the app.

2. Would it be possible to query and display the volt and ampere values ​​of the two MPPT trackers?

As I understand the parameter list, it should be available in the interface:
- _mppt.inAmp_ | int | PV1 input current (mA)
- _mppt.inVol_ | int | PV1 input voltage (mV)
- _mppt.pv2InAmp_ | int | PV2 input current: The current voltage is amplified 100 times.
- _mppt.pv2InVol_ | int | PV2 input voltage: The actual voltage is amplified 10 times.
- source: [https://developer-eu.ecoflow.com/us/document/delta2max](url)

And one more little thing, that would also be interesting :-)
- _bms_emsStatus.fanLevel_ | int | Fan level

I hope someone can take care of this and find a solution.

**Comment by giovanne123** [comment]
> Hi, I have two questions/ideas about the parameters of the Delta 2 Max.
> 
> 1. Do I understand correctly that you can set the AC charging speed using the entity _number.delta_2_max_ac_charging_power_ and does that work for you? I always get 2400W displayed in the app.

Have you the switch on the backside of the D2M in the right position?

**Reply by Neewieedo** [reply]
Thank you for the tip! The switch was set to fast.
I didn't even know that this switch existed :-)

I set the switch to slow and now I can set the AC charging value

**Comment by Neewieedo** [comment]
> 2\. Would it be possible to query and display the volt and ampere values ​​of the two MPPT trackers?
> 
> As I understand the parameter list, it should be available in the interface:
> 
> * _mppt.inAmp_ | int | PV1 input current (mA)
> * _mppt.inVol_ | int | PV1 input voltage (mV)
> * _mppt.pv2InAmp_ | int | PV2 input current: The current voltage is amplified 100 times.
> * _mppt.pv2InVol_ | int | PV2 input voltage: The actual voltage is amplified 10 times.
> * source: [https://developer-eu.ecoflow.com/us/document/delta2max](https://developer-eu.ecoflow.com/us/document/delta2max)

Is there an easy way to query the values ​​via the http interface for testing purposes?

**Comment by felinfurtif** [comment]
Hi,
I'm posting here since my questions are about Delta 2 Max parameters. Let me know if I should move this elsewhere.
I'm a beginner with Home Assistant, and just integrated my Delta 2 Max. Thank you very much for making this possible.

So, I'm trying to set up a simple automation where the trigger is the battery level. 

1st question : what's the difference between entities "battery_level" and "main_battery_level" ? 
I can see the latter has more state attributes, but I'm not sure which one I should use for the automation.

2nd question : why is HA not readily reading values from these entities? It just reads "unknown". How can I fix this? 

With this I'll be on my way to my first automation ever!

**Comment by Neewieedo** [comment]
Are the solar inputs of the battery always switched off when the maximum charge level is reached?
Can't the generated solar power simply be passed on 1:1 to the PowerStream?

**Comment by Neewieedo** [comment]
@tolwi Is it possible to add these parameters to the Delta 2 (Max)?

> * _mppt.inAmp_ | int | PV1 input current (mA)
> * _mppt.inVol_ | int | PV1 input voltage (mV)
> * _mppt.pv2InAmp_ | int | PV2 input current: The current voltage is amplified 100 times.
> * _mppt.pv2InVol_ | int | PV2 input voltage: The actual voltage is amplified 10 times.
> * source: [https://developer-eu.ecoflow.com/us/document/delta2max](https://developer-eu.ecoflow.com/us/document/delta2max)

**Comment by Neewieedo** [comment]
#336 
Big Thanks!!
